### ISSUE TITLE
README improvement to OK.success and OK.failure

### ISSUE THREAD

**Issue by seymores** [CrowdHailer/OK#43]
I wish the result of OK.success and OK.failure is clarified in the doc as I spent some time trying to understand what the result of these function calls are.

And a simple one line doc explaining the result in the code sample should suffice.

```elixir
require OK

OK.with do
  user <- fetch_user(1)
  cart <- fetch_cart(1)
  order = checkout(cart, user)
  saved <- save_order(order)
  OK.success saved   # returns {:ok, saved}
else
  :user_not_found ->
    OK.failure :unauthorized   # returns {:error, :unauthorized}
end
```

**Comment by CrowdHailer** [comment]
Looks good to me :+1:

**Comment by CrowdHailer** [comment]
Hey if you want to add this as PR that would be great

**Comment by CrowdHailer** [comment]
closed by #46

**Comment by CrowdHailer** [comment]
Changes should be in 1.9.4 docs
